experimental/examples/tree-sample/inventory-app/src/views/main.tsx

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISharedTree } from "@fluid-internal/tree";
+import * as React from "react";
+import { useTree } from "@fluid-experimental/tree-react-api";
+import { Inventory } from "../domains/model";
+import { Counter } from "./counter";
+
+export const MainView: React.FC<{ tree: ISharedTree }> = ({ tree }) => {
+	const inventory = useTree<Inventory>(tree);
+
+	const counters = Object.keys(inventory).map((key) => {
+		const value = inventory[key] as number;
+
+		return (
+			<Counter
+				title={key}
+				count={value}
+				decrementButtonProps={{
+					title: "-",
+					disabled: !(value > 0),
+					onClick: () => (inventory[key] as number)--,
+				}}
+				incrementButtonProps={{
+					title: "+",
+					disabled: false,
+					onClick: () => (inventory[key] as number)++,
+				}}
+			></Counter>
```

Comment:
Having `MainView` pass `decrementButtonProps`/`incrementButtonProps` to `Counter` means `MainView` must have intimate knowledge of `Counter`'s construction.  Consider something like this instead, leaving the disable logic and button labeling to the `Counter`:

```suggestion
			<Counter
				title={key}
				count={value}
				decrement={ () => (inventory[key] as number)-- }
				increment={ () => (inventory[key] as number)++ }
			></Counter>
```

If it's critical to protect against negative values, this logic should probably be enforced in the schema (can it do this?) or else in the `DataObject` holding the tree (though this would preclude handing the tree directly to the view) rather than in the view.